### ISSUE TITLE
[Xcode4 static analyzer] NSMakeCollectable misuse (w/fix)

### ISSUE THREAD

**Issue by AliSoftware** [pokeb/asi-http-request#222]
In your code, you are using NSMakeCollectable this way:

```
NSMakeCollectable( [ ((NSString*)cfstr) autorelease] );
```

Whereas according to [the documentation](http://developer.apple.com/library/mac/#documentation/Cocoa/Conceptual/GarbageCollection/Articles/gcCoreFoundation.html), you should do the opposite, calling NSMakeCollectable first and autorelease afterwards:

```
[NSMakeCollectable(cfstr) autorelease];
```

Casting the CFTypeRef to a NSObject first, calling autorelease, and then calling NSMakeCollectable seems, according to the static analyzer, to make the object receive an additional autorelease message. Inverting the calls as suggested in the documentation removes the clang warning.

**Comment by jameswomack** [comment]
I've been seeing this same issue

**Comment by pokeb** [comment]
I merged #224 which should correct this.
